Code Highlighting and Deployment Issues

mansona opened the pull request:
- Supercedes https://github.com/ember-learn/guides-app/pull/7 in adding code highlighting 
- adding ember-cli-deploy config 
  - currently configured to deploy to Stone Circle S3 for demo
- fixing most of the app for use in fastboot
- adding prember
- adding favicon
- forcing all urls to end in `/`
- fixing # based scrolling in the deployed app 🎉

jenweber commented:
Skimmed the code. Ran it locally. Generally looks good!

It looks like you lost the diff highlighting:
![screen shot 2018-01-20 at 6 49 42 pm](https://user-images.githubusercontent.com/16627268/35189152-c57f0566-fe12-11e7-8b83-e17ea731a0fc.png)

Production ember.js site
![screen shot 2018-01-20 at 6 49 53 pm](https://user-images.githubusercontent.com/16627268/35189153-c772b1c4-fe12-11e7-9f42-2eacaabcb8d7.png)


Some things I noticed to fix in future PRs:
- "edit" links are not functional
- using nav links at the bottom doesn't bring you back up to the top on the next page
- navigating straight to a link with a `/` at the end (or anchor tag) makes the article header disappear, but now links still work on the page with a trailing `/` 🎉

mansona commented:
Thanks @jenweber 🎉  Now that I've rebased this I have lost the order of what things were fixed before or after your review 😖 I will hit your points one by one and see where that gets us

- diff highlighting
	- see https://github.com/ember-learn/guides-app/issues/21
	- I was under the impression that we are waiting until after the first release for adding that back in. we can discuss
- "edit" links are not functional
	- fixed ✅ 
- using nav links at the bottom doesn't bring you back up to the top on the next page
	- fixed ✅ 
- navigating straight to a link with a / at the end (or anchor tag) makes the article header disappear
	- fixed ✅

mansona commented:
@jenweber those things I mentioned in my last comment are now deployed on http://emberguides.stonecircle.io so you can verify them if you would like 👍
